ocode_python/core/semantic_context.py

```
@@ -0,0 +1,779 @@
+        if cache_key in self.expansion_cache:
+            additional_paths = self.expansion_cache[cache_key]
+        else:
+            # Scan for additional relevant files
+            all_files = await self.base_context_manager.scan_project()
+            current_paths = set(current_context.files.keys())
+            candidate_files = [f for f in all_files if f not in current_paths]
+
+            # Read candidate files
+            candidate_content = {}
+            for file_path in candidate_files[:50]:  # Limit for performance
+                content = await self.base_context_manager._read_file(file_path)
```

Pylint (3.3.7)

[warning] Access to a protected member _read_file of a client class

(W0212)